File: android/rctmapboxar/src/main/java/com/mapbox/react/ar/utils/ListUtils.java
package com.mapbox.react.ar.utils;

import java.util.List;

public class ListUtils {
    public static String join(List<String> list, String delimiter) {
        StringBuilder builder = new StringBuilder();

        for (int i = 0; i < list.size(); i++) {
            String item = list.get(i);

            if (i == list.size() - 1) {
                builder.append(item);
            } else {
                builder.append(item + delimiter);
            }
        }

        return builder.toString();
    }
}

File: ios/RNMapboxAR/ListUtils.m
//
//  ListUtils.m
//  RNLocateTest
//
//  Created by Dave Prukop on 5/4/18.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "ListUtils.h"

@implementation ListUtils

+ (NSString *)join:(NSArray<NSString *> *)list withDelimiter:(NSString *)delimiter
{
  NSMutableString *builder = [[NSMutableString alloc] init];
  
  for (int i=0; i<[list count]; i++) {
    NSString *item = [list objectAtIndex:i];
    
    if (i == [list count] - 1) {
      [builder appendString:item];
    } else {
      [builder appendString:[NSString stringWithFormat:@"%@%@", item, delimiter]];
    }
  }
       
  return [NSString stringWithString:builder];
}

@end
